plugin/storage/es/factory.go

```diff
@@ -236,8 +236,8 @@ func createSpanReader(
 		MaxDocCount:                   cfg.MaxDocCount,
 		MaxSpanAge:                    cfg.MaxSpanAge,
 		IndexPrefix:                   cfg.IndexPrefix,
-		SpanIndexDateLayout:           cfg.IndexDateLayoutSpans,
-		ServiceIndexDateLayout:        cfg.IndexDateLayoutServices,
+		SpanIndexDateLayout:           cfg.Indices.Spans.DateLayout,
```

Comment:
it looks like SpanReader needs settings for more than one index. Instead of denormalizing the new structs that you introduced in pkg/es/config/config.go, can we just reuse them as part of SpanReaderParams?  There's no need to keep having these different / inconsistent data models in different layers. And you'd just be able to assign that struct by value here.